modules/objdetect/include/opencv2/objdetect/face.hpp

```diff
@@ -0,0 +1,82 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef _OPENCV_DNN_FACE_CORE_HPP_
+#define _OPENCV_DNN_FACE_CORE_HPP_
+
+#include <opencv2/core.hpp>
+
+/** @defgroup dnn_face DNN-based face detection and recognition
+ */
+
+namespace cv
+{
+
+/** @brief DNN-based face detector, model download link: https://github.com/ShiqiYu/libfacedetection.train/tree/master/tasks/task1/onnx.
+ */
+class CV_EXPORTS_W DNNFaceDetector
+{
+public:
+    virtual ~DNNFaceDetector() {};
+
+    /** @brief A simple interface to detect face from given image
+     *
+     *  @param image an image to detect
+     *  @param faces detection results stored in a cv::Mat
+     */
+    CV_WRAP virtual int detect(InputArray image, OutputArray faces) = 0;
+
+    /** @brief Creates an instance of this class with given parameters
+     * 
+     *  @param onnx_path the path to the downloaded ONNX model
+     *  @param input_size the size of the input image
+     *  @param score_threshold the threshold to filter out bounding boxes of score smaller than the given value
+     *  @param nms_threshold the threshold to suppress bounding boxes of IoU bigger than the given value
+     *  @param top_k keep top K bboxes before NMS
+     *  @param backend_id the id of backend
+     *  @param target_id the id of target device
+     */
+    CV_WRAP static Ptr<DNNFaceDetector> create(const String& onnx_path,
+                                                const Size& input_size,
```

Comment:
1. there must be 2 filenames specified, and onnx_ prefix should not be used. We should not give any preference to one particular network.

2. I think, there should be properties to adjust input size, as well as the score & nms thresholds, as well as top_k parameters dynamically, to achieve the optimal speed/quality ratio.

Reply:
> 1. there must be 2 filenames specified, and onnx_ prefix should not be used. We should not give any preference to one particular network.

Do you mean something like this?
https://github.com/opencv/opencv/blob/b61a55eebf1e654106bde3222e926d0e5b8edbca/modules/dnn/include/opencv2/dnn/dnn.hpp#L1514-L1516

I was thinking our target was a specific implementation on our face detection model. It is not feasible to make it a one-for-all implementation since different models have different settings in prior boxes and different encoding & decoding processes.

Reply:
Done for 1. at https://github.com/opencv/opencv/pull/20422/commits/dbb9ea3d04d36a501223d1c3253cc20d015ca71b.